### PR TITLE
feat: Integrate SIP-12 update

### PR DESCRIPTION
**Pull request by hmalik88**
This PR brings the domain resolution integration up to specification.

There's a lot going on in this PR so here's a quick breakdown of what's going on:

- The `DomainInputResolutionCell` component is responsible for displaying a single resolution, it can properly display both ENS and snap provided resolutions. It also has logic to detect overflowing titles so that it can replace the title component with one that has an ellipsis and a tooltip showing the entire title.
- The confusable component was modified to include an option to wrap the points in a text component instead of span and I added a prop to be able to assign a classname to the tooltip wrapper, I did this to allow for my use case of having a tooltip within a tooltip (an overflowing domain name with confusables).
- The update to the `AddressListItem` component snapshot is because of the update to the confusable component.
- ~~I had to update implementation of the `AddRecipient` components because the update in the name resolution api would break the existing implementation. Those components will be removed at some point by the extension team.~~ (These components have since been removed)
- The `AddContact` component was updated to use the domain input resolution cell component.
- The `AddressListItem` component was updated to accommodate overflowing titles (specifically when viewing all contacts).
- The `ViewContact` & `EditContact` components were also updated to accommodate overflowing titles.
- After the add/remove from address book actions, the old state would momentarily hang, so now I’m forcing the update for a cleaner UX where the old state won’t hang at all.

**Previous `AddContact` screen**: 

https://github.com/MetaMask/metamask-extension/assets/41640681/ccaebf23-76c9-4ae8-99ee-f338d9da669d

**New screens**:

https://github.com/MetaMask/metamask-extension/assets/41640681/957def02-5859-4299-8c21-b561ea8f3c2f

Note: The `AddContact` screen was re-done to accommodate for snap provided resolutions and to update the designs as it seemed the screen was largely undesigned. See previous design video.

### CODE REVIEW

**Comment by Montoya** [comment]
Updated the PR with the correct video for the new UX

**Review by FrederikBolding — COMMENTED** [review]
Not sure if related to this branch or something else, but the send flow is performing really badly on my end.

File: ui/ducks/send/send.js [hunk 1]
```diff
@@ -2260,9 +2260,9 @@ export function resetRecipientInput() {
     const state = getState();
     const chainId = getCurrentChainId(state);
     await dispatch(addHistoryEntry(`sendFlow - user cleared recipient input`));
+    await dispatch(resetDomainResolution());
```
Comment (FrederikBolding) [inline ui/ducks/send/send.js:2263]: ?

File: ui/pages/confirmations/send/send-content/add-recipient/add-recipient.container.test.js [hunk 2]
```diff
@@ -48,9 +52,13 @@ describe('add-recipient container', () => {
     it('should map the correct properties to props', () => {
       expect(mapStateToProps('mockState')).toStrictEqual({
         addressBook: [{ name: 'mockAddressBook:mockState' }],
-        addressBookEntryName: undefined,
         contacts: [{ name: 'mockAddressBook:mockState' }],
-        domainResolution: 'mockSendDomainResolution:mockState',
+        domainResolutions: [
+          {
+            resolvedAddress: 'mockSendDomainResolutions:mockState',
+            protocol: 'Ethereum Name Service',
```
Comment (FrederikBolding) [inline ui/pages/confirmations/send/send-content/add-recipient/add-recipient.container.test.js:59]: Missing `addressBookEntryName` and  `resolvingSnap`

File: ui/components/multichain/pages/send/components/recipient.tsx [hunk 3]
```diff
@@ -58,15 +58,16 @@ export const SendPageRecipient = () => {
   const recipient = useSelector(getRecipient);
   const userInput = useSelector(getRecipientUserInput) || '';
 
-  const domainResolution = useSelector(getDomainResolution);
+  const domainResolutions = useSelector(getDomainResolutions) || [];
   const domainError = useSelector(getDomainError);
   const domainWarning = useSelector(getDomainWarning);
 
   let addressBookEntryName = '';
+  const domainResolution = domainResolutions[0];
```
Comment (FrederikBolding) [inline ui/components/multichain/pages/send/components/recipient.tsx:66]: ? 🤔

File: ui/components/ui/confusable/confusable.component.js [hunk 4]
```diff
@@ -1,39 +1,48 @@
...
+  return confusableData.map(({ point, similarTo }) => {
     const zeroWidth = similarTo === '';
     if (similarTo === undefined) {
-      return point;
+      return asText ? <Text key={uuidv4()}>{point}</Text> : point;
     }
     return (
       <Tooltip
-        key={index.toString()}
+        key={uuidv4()}
```
Comment (FrederikBolding) [inline ui/components/ui/confusable/confusable.component.js:22]: Why is this necessary now?

File: ui/ducks/domains.js [hunk 5]
```diff
@@ -194,7 +192,24 @@ export async function fetchResolutions({ domain, chainId, state }) {
...
+
+    if (lookupMatchersCaveat) {
+      const { tlds, schemes } = lookupMatchersCaveat;
+      if (
+        tlds?.some((tld) => domain.endsWith(`.${tld}`)) ||
+        schemes?.some((scheme) => domain.startsWith(`${scheme}:`))
+      ) {
+        return true;
+      }
+      return false;
```
Comment (FrederikBolding) [inline ui/ducks/domains.js:209]:
```suggestion
     return tlds?.some((tld) => domain.endsWith(`.${tld}`)) ||
        schemes?.some((scheme) => domain.startsWith(`${scheme}:`))
```

File: ui/pages/confirmations/send/send-content/add-recipient/domain-input-resolution-cell.js [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+  useEffect(() => {
+    let isOverflowing =
+      titleRef.current.offsetWidth < titleRef.current.scrollWidth;
+    const breakpointLength = titleRef.current.textContent.length;
+
+    if (isOverflowing && !breakpointRef.current) {
+      breakpointRef.current = breakpointLength;
+    }
+
+    if (!isOverflowing) {
```
Comment (FrederikBolding) [inline ui/pages/confirmations/send/send-content/add-recipient/domain-input-resolution-cell.js:61]: What is going on here

File: ui/pages/confirmations/send/send-content/add-recipient/domain-input-resolution-cell.js [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+          className="send__select-recipient-wrapper__group-item__title"
+        >
+          <Title />
+        </div>
+        {domainName && (
+          <div className="send__select-recipient-wrapper__group-item__subtitle">
+            <Text color={TextColor.textAlternative}>{ellipsify(address)}</Text>
+          </div>
+        )}
+        {domainType === 'ENS' && (
```
Comment (FrederikBolding) [inline ui/pages/confirmations/send/send-content/add-recipient/domain-input-resolution-cell.js:181]: Will this ever not be the case?

**Review by FrederikBolding — COMMENTED** [review]
Please write an elaborate description detailing the changes here, there is a lot of changes and not a lot of context

File: ui/ducks/send/send.js (hunk 1 above)
Comment (hmalik88) [reply ui/ducks/send/send.js:2263]: Had to add this in to clear resolution when hitting the x to clear input in the send screen. Otherwise the resolution would hang.

File: ui/pages/confirmations/send/send-content/add-recipient/domain-input-resolution-cell.js (hunk 7 above)
Comment (hmalik88) [reply ui/pages/confirmations/send/send-content/add-recipient/domain-input-resolution-cell.js:181]: No, because there isn't always a domain resolution, a hex address can be typed in.

File: ui/components/ui/confusable/confusable.component.js (hunk 4 above)
Comment (hmalik88) [reply ui/components/ui/confusable/confusable.component.js:22]: using just the index can mean that with multiple entries multiple points can have the same key

File: ui/pages/confirmations/send/send-content/add-recipient/domain-input-resolution-cell.js (hunk 6 above)
Comment (hmalik88) [reply ui/pages/confirmations/send/send-content/add-recipient/domain-input-resolution-cell.js:61]: This logic is to ensure that the cell properly shows the ellipsis in the domain resolution cell

File: ui/components/multichain/pages/send/components/recipient.tsx (hunk 3 above)
Comment (hmalik88) [reply ui/components/multichain/pages/send/components/recipient.tsx:66]: Hmm, missed this component being used anywhere before so I was updating it so tests would pass. Idk how I should proceed here tbh because the design they use for their `AddressListItem` component is divergent from ours.

Comment (FrederikBolding) [reply ui/components/multichain/pages/send/components/recipient.tsx:66]: Perhaps we need to loop in @darkwing

File: ui/pages/confirmations/send/send-content/add-recipient/add-recipient.container.test.js (hunk 2 above)
Comment (hmalik88) [reply ui/pages/confirmations/send/send-content/add-recipient/add-recipient.container.test.js:59]: Resolving this because both entries can be `undefined` and in this scenario it makes sense to leave out.

File: app/_locales/el/messages.json [hunk 8]
```diff
@@ -5220,7 +5220,8 @@
     "message": "Υποβλήθηκε"
   },
   "suggestedBy": {
-    "message": "Προτείνεται από"
+    "message": "Προτείνεται από",
+    "description": "$1 is the snap name"
```
[the remark this answers is not in the document]
Comment (hmalik88) [reply app/_locales/el/messages.json:5224]: yeah idk what's going on here, these changes came in from develop hmm

[the remark this answers is not in the document]
Comment (hmalik88) [reply app/_locales/el/messages.json]: ok restored as per develop [7de69ac](https://github.com/MetaMask/metamask-extension/pull/23697/commits/7de69ac1df621848d4286516b09994b0b1aba1a6)

File: ui/components/component-library/badge-wrapper/badge-wrapper.types.ts [hunk 9]
```diff
@@ -46,6 +46,9 @@ export interface BadgeWrapperStyleUtilityProps extends StyleUtilityProps {
     right?: number | string;
     bottom?: number | string;
     left?: number | string;
+    transform?: string;
+    width?: string;
+    height?: string;
```
Comment (georgewrmarshall) [inline ui/components/component-library/badge-wrapper/badge-wrapper.types.ts:51]: Instead of adding these additional prop types, can we add a custom class name to the badge container via the `badgeContainerProps` object? I noticed that we could make some improvements to the `BadgeWrapper` component on line `37`:

```jsx
<Box
  {...badgeContainerProps} // won't overwrite className prop if `badgeContainerProps.className` exists
  className={classnames(
    'mm-badge-wrapper__badge-container',
    {
      [`mm-badge-wrapper__badge-container--${anchorElementShape}-${position}`]: !positionObj,
    },
    badgeContainerProps?.className || '', // custom class name which can contain the transform styles
  )}
  style={{ ...positionObj }}
>
```

File: ui/components/multichain/pages/send/components/domain-input-resolution-cell.tsx [hunk 10]
```diff
@@ -0,0 +1,219 @@
...
+DomainInputResolutionCell.propTypes = {
+  domainType: PropTypes.string.isRequired,
+  address: PropTypes.string.isRequired,
+  domainName: PropTypes.string.isRequired,
+  ///: BEGIN:ONLY_INCLUDE_IF(build-flask)
+  resolvingSnap: PropTypes.string.isRequired,
+  ///: END:ONLY_INCLUDE_IF
+  onClick: PropTypes.func,
+  protocol: PropTypes.string,
+};
```
Comment (georgewrmarshall) [inline ui/components/multichain/pages/send/components/domain-input-resolution-cell.tsx:219]: Would make it much easier for me to review if this component had a storybook story. Right now I have to install flask, install the snap, rebuild the extension with correct env flags etc... Instead if there was a story I could use the storybook build in the PR

**Review by georgewrmarshall — COMMENTED** [review]
Hey @hmalik88, this is looking good. The `positionObj` wasn't intended to provide much more functionality that simple positioning. If you require further style customization I'd suggest adding a custom className and styles. That way we don't have to adjust the prop object for every new style definition

File: ui/components/multichain/pages/send/components/domain-input-resolution-cell.tsx (hunk 10 above)
Comment (hmalik88) [reply ui/components/multichain/pages/send/components/domain-input-resolution-cell.tsx:219]: Apologies! I just pushed up a stories file

File: ui/components/component-library/badge-wrapper/badge-wrapper.types.ts (hunk 9 above)
Comment (hmalik88) [reply ui/components/component-library/badge-wrapper/badge-wrapper.types.ts]: Done: [c41aa39](https://github.com/MetaMask/metamask-extension/pull/23697/commits/c41aa39552509a3bcf101ca7ac9864d653777ddf)

File: ui/pages/settings/contact-list-tab/add-contact/add-contact.component.js [hunk 11]
```diff
@@ -134,6 +127,43 @@ export default class AddContact extends PureComponent {
               {t('ethereumPublicAddress')}
             </div>
             {this.renderInput()}
+            <div
+              className={`address-book__view-contact__group__${
+                domainResolutions?.length === 1 ? 'single-' : ''
+              }resolution-list`}
+            >
+              {domainResolutions?.length > 0 &&
+                domainResolutions.map((resolution) => {
```
Comment (darkwing) [inline ui/pages/settings/contact-list-tab/add-contact/add-contact.component.js:136]: Can we remove the `domainResolutions?.length` check and just do `domainResolutions?.map`?  Mapping over empty would be fine.

File: ui/pages/settings/contact-list-tab/add-contact/add-contact.component.js [hunk 12]
```diff
@@ -134,6 +127,43 @@ export default class AddContact extends PureComponent {
...
+                    resolvingSnap,
+                    addressBookEntryName,
+                    protocol,
+                  } = resolution;
+                  const domainName = addressBookEntryName || this.state.input;
+                  return (
+                    <DomainInputResolutionCell
+                      key={`${resolvedAddress}${resolvingSnap}${protocol}`}
+                      domainType={
+                        protocol === 'Ethereum Name Service' ? 'ENS' : 'Other'
```
Comment (darkwing) [inline ui/pages/settings/contact-list-tab/add-contact/add-contact.component.js:148]: Should ENS be translated?  I'm not sure...

File: ui/components/ui/confusable/confusable.component.js [hunk 13]
```diff
@@ -1,39 +1,48 @@
...
             : t('confusableUnicode', [point, similarTo])
         }
+        wrapperClassName={confusableWrapperName}
       >
-        <span className="confusable__point">{zeroWidth ? '?' : point}</span>
+        {asText ? (
+          <Text className="confusable__point">{zeroWidth ? '?' : point}</Text>
+        ) : (
+          <span className="confusable__point">{zeroWidth ? '?' : point}</span>
+        )}
```
Comment (darkwing) [inline ui/components/ui/confusable/confusable.component.js:36]: To shorten this I think it can be:

```
<Text className="confusable__point" as={asText ? 'p': 'span'}>{zeroWidth ? '?' : point}</Text>

```

File: ui/components/multichain/pages/send/components/domain-input-resolution-cell.tsx [hunk 14]
```diff
@@ -0,0 +1,219 @@
...
+        confusableWrapperName="multichain-send-page__recipient__item__title-confusable-wrapper"
+      />
+    </Tooltip>
+  );
+
+  ///: BEGIN:ONLY_INCLUDE_IF(build-flask)
+  if (domainType === 'Other') {
+    // Snap provided resolution.
+    return (
+      <div
```
Comment (darkwing) [inline ui/components/multichain/pages/send/components/domain-input-resolution-cell.tsx:110]: Instead of using `<div>`, can we use `<Box>` instead?  That's been our practice for a year or so now.

Comment (darkwing) [reply ui/components/multichain/pages/send/components/domain-input-resolution-cell.tsx:110]: Also, could the be separated into its own component?  Seems a decent sized alternative block.

File: ui/pages/settings/contact-list-tab/add-contact/add-contact.component.js (hunk 12 above)
Comment (hmalik88) [reply ui/pages/settings/contact-list-tab/add-contact/add-contact.component.js:148]: Going by our designs here: https://www.figma.com/design/7JOVfBCxfHiz8xOwbuWF3j/Snaps-Platform-v2?node-id=69-16627&t=CxxNxGwGA8BVPq1s-0


cc: @eriknson -- could use your comment on this part as well, I know @georgewrmarshall already tagged you about the icon size.

File: ui/components/multichain/pages/send/components/domain-input-resolution-cell.tsx (hunk 14 above)
Comment (hmalik88) [reply ui/components/multichain/pages/send/components/domain-input-resolution-cell.tsx:110]: I'd like to keep as is, I think it would increase complexity to make a distinction between `ENS` and snap related resolution components, especially when the ENS resolution is being planned to be rolled into a snap in the future.

File: ui/ducks/send/send.js [hunk 15]
```diff
@@ -2740,15 +2740,15 @@ export function useMyAccountsForRecipientSearch() {
...
+  return (dispatch, getState) => {
     const state = getState();
     const chainId = getCurrentChainId(state);
     showLoadingIndication();
-    await dispatch(addHistoryEntry(`sendFlow - user cleared recipient input`));
-    await dispatch(updateRecipientUserInput(''));
-    await dispatch(updateRecipient({ address: '', nickname: '' }));
-    await dispatch(resetDomainResolution());
-    await dispatch(validateRecipientUserInput({ chainId }));
+    dispatch(addHistoryEntry(`sendFlow - user cleared recipient input`));
```
Comment (FrederikBolding) [inline ui/ducks/send/send.js:2747]: Why do we not need to await here anymore?

File: ui/pages/settings/contact-list-tab/index.scss
Comment (FrederikBolding) [inline ui/pages/settings/contact-list-tab/index.scss:1]: image
This looks a bit off

File: ui/components/multichain/pages/send/components/domain-input-resolution-cell.tsx [hunk 16]
```diff
@@ -0,0 +1,244 @@
...
+          isOverflowing = true;
+        } else {
+          isOverflowing = false;
+          breakpointRef.current = null;
+        }
+      }
+    }
+
+    if (isOverflowing !== isTitleOverflowing) {
+      setIsTitleOverflowing(isOverflowing);
```
Comment (FrederikBolding) [inline ui/components/multichain/pages/send/components/domain-input-resolution-cell.tsx:90]: What is the goal of all of this?

File: ui/pages/settings/contact-list-tab/add-contact/add-contact.component.js [hunk 17]
```diff
@@ -134,6 +127,40 @@ export default class AddContact extends PureComponent {
...
+                    domainName={domainName}
+                    onClick={() => {
+                      this.setState({
+                        selectedAddress: resolvedAddress,
+                        newName: this.state.newName || domainName,
+                      });
+                      this.props.resetDomainResolution();
+                    }}
+                    protocol={protocol}
+                    resolvingsnap={resolvingSnap}
```
Comment (FrederikBolding) [inline ui/pages/settings/contact-list-tab/add-contact/add-contact.component.js:159]:
```suggestion
                    resolvingSnap={resolvingSnap}
```

File: ui/components/multichain/pages/send/components/recipient.tsx [hunk 18]
```diff
@@ -10,93 +10,98 @@ import {
...
-
 export const SendPageRecipient = () => {
   const t = useContext(I18nContext);
   const dispatch = useDispatch();
 
   const recipient = useSelector(getRecipient);
   const userInput = useSelector(getRecipientUserInput) || '';
 
-  const domainResolution = useSelector(getDomainResolution);
+  const domainResolutions = useSelector(getDomainResolutions) || [];
```
Comment (FrederikBolding) [inline ui/components/multichain/pages/send/components/recipient.tsx:36]:
```suggestion
  const domainResolutions = useSelector(getDomainResolutions) ?? [];
```

Comment (FrederikBolding) [reply ui/components/multichain/pages/send/components/recipient.tsx:36]: But why is this needed in the first place? 🤔

File: ui/ducks/send/send.js (hunk 15 above)
Comment (hmalik88) [reply ui/ducks/send/send.js:2747]: I'm not sure why we were in the first place? None of these actions are async. Regardless, this was causing an async issue where the send page was using old state, specifically line 103 in `Recipient.tsx`: `userInput.length > 0 ? CONTACTS_TAB_KEY : ACCOUNTS_TAB_KEY`

File: ui/components/multichain/pages/send/components/domain-input-resolution-cell.tsx (hunk 16 above)
Comment (hmalik88) [reply ui/components/multichain/pages/send/components/domain-input-resolution-cell.tsx:90]: Trying to avoid setting state when it's not necessary

File: ui/components/multichain/pages/send/components/recipient.tsx (hunk 18 above)
Comment (hmalik88) [reply ui/components/multichain/pages/send/components/recipient.tsx:36]: initial state for domain resolutions is `null`.

File: ui/pages/settings/contact-list-tab/index.scss
Comment (hmalik88) [reply ui/pages/settings/contact-list-tab/index.scss:1]: `AddressListItem` component issue, fixed here: [168888d](https://github.com/MetaMask/metamask-extension/pull/23697/commits/168888d9f28813dbac682899ad61ef5dbb649039)

File: ui/components/multichain/pages/send/components/domain-input-resolution-cell.tsx (hunk 16 above)
Comment (FrederikBolding) [reply ui/components/multichain/pages/send/components/domain-input-resolution-cell.tsx:90]: Apologies, I should have clarified. What are we trying to achieve by detecting the overflow and using different titles?

File: ui/ducks/send/send.js (hunk 15 above)
Comment (FrederikBolding) [reply ui/ducks/send/send.js:2747]: Looks like `updateRecipient` is async?

File: ui/components/multichain/pages/send/components/recipient.tsx [hunk 19]
```diff
@@ -10,93 +10,98 @@ import {
...
+  } else if (domainResolutions?.length > 0 && !recipient.error) {
+    contents = domainResolutions.map((domainResolution: DomainResolution) => {
+      const { resolvedAddress, resolvingSnap, addressBookEntryName, protocol } =
+        domainResolution;
+      return (
+        <DomainInputResolutionCell
+          key={`${resolvedAddress}${resolvingSnap}${protocol}`}
+          domainType={domainType}
+          address={resolvedAddress}
+          domainName={addressBookEntryName || userInput}
```
Comment (FrederikBolding) [inline ui/components/multichain/pages/send/components/recipient.tsx:86]:
```suggestion
          domainName={addressBookEntryName ?? userInput}
```

File: ui/components/multichain/pages/send/components/domain-input-resolution-cell.tsx (hunk 16 above)
Comment (hmalik88) [reply ui/components/multichain/pages/send/components/domain-input-resolution-cell.tsx:90]: Oh, so that we can display the overflowing title in a tooltip. We want to display the whole resolved name because there are snaps that will fuzzy match, so what you're typing in is not necessarily what you're getting in the resolution.

File: ui/ducks/send/send.js (hunk 15 above)
Comment (hmalik88) [reply ui/ducks/send/send.js:2747]: Woops didn't realize, well awaiting just that seems to be fine as well. Pushing an update.

File: ui/pages/settings/contact-list-tab/edit-contact/edit-contact.component.js [hunk 20]
```diff
@@ -74,13 +75,21 @@ export default class EditContact extends PureComponent {
...
+            alignItems={AlignItems.center}
+            style={{ overflow: 'hidden' }}
+            paddingRight={2}
+          >
             <AvatarAccount size={AvatarAccountSize.Lg} address={address} />
             <Text
               className="address-book__header__name"
               variant={TextVariant.bodyLgMedium}
               marginInlineStart={4}
+              style={{ overflow: 'hidden', textOverflow: 'ellipsis' }}
```
Comment (georgewrmarshall) [inline ui/pages/settings/contact-list-tab/edit-contact/edit-contact.component.js:92]: Can you use the ellipsis prop here? https://metamask.github.io/metamask-storybook/?path=/docs/components-componentlibrary-text--docs#ellipsis

Screenshot 2024-06-06 at 12 31 14 PM

```suggestion
              ellipsis
```

File: ui/pages/settings/contact-list-tab/view-contact/view-contact.component.js [hunk 21]
```diff
@@ -51,12 +51,14 @@ function ViewContact({
...
           className="settings-page__header address-book__header"
           paddingLeft={6}
+          paddingRight={6}
         >
           <AvatarAccount size={AvatarAccountSize.Lg} address={address} />
           <Text
             className="address-book__header__name"
             variant={TextVariant.bodyLgMedium}
             marginInlineStart={4}
+            style={{ overflow: 'hidden', textOverflow: 'ellipsis' }}
```
Comment (georgewrmarshall) [inline ui/pages/settings/contact-list-tab/view-contact/view-contact.component.js:61]: Sam here 
```suggestion
            ellipsis
```

**Review by FrederikBolding — DISMISSED** [review]

**Review by FrederikBolding — DISMISSED** [review]

File: ui/pages/settings/contact-list-tab/edit-contact/edit-contact.component.js [hunk 22]
```diff
@@ -74,27 +75,38 @@ export default class EditContact extends PureComponent {
...
-            className="settings-page__address-book-button"
-            onClick={async () => {
-              await removeFromAddressBook(chainId, address);
-              history.push(listRoute);
-            }}
-          >
-            {t('deleteContact')}
-          </Button>
+          <Box className="settings-page__address-book-button">
+            <Button
```
Comment (georgewrmarshall) [inline ui/pages/settings/contact-list-tab/edit-contact/edit-contact.component.js:99]: non-blocking: it would help if we can replace deprecated code with component-library in files that are updated

Screenshot 2024-06-11 at 8 27 36 AM

**Review by georgewrmarshall — DISMISSED** [review]
LGTM! UI is looking great 💯  a few non-critical things to remember that we should probably add some linting for
- We should try to avoid inline styles in favor of custom CSS classnames
- It would be great if we can replace deprecated components in files that are updated

File: test/e2e/tests/settings/address-book.spec.js [hunk 23]
```diff
@@ -166,12 +167,12 @@ describe('Address Book', function () {
...
-        await driver.clickElement({ text: 'Delete contact', tag: 'a' });
-
-        const contact = await driver.findElement('.address-list-item__label');
+        await driver.clickElement('.settings-page__address-book-button');
 
         // it checks if account is deleted
-        const exists = await driver.isElementPresent(contact);
+        const exists = await driver.isElementPresent(
+          By.css('.address-list-item__label'),
+        );
```
[the remark this answers is not in the document]
Comment (hmalik88) [reply test/e2e/tests/settings/address-book.spec.js:175]: `findElement` was hanging and so I had to use `isElementPresent` and the `By` class, I don't see any static method that lets me select with data-test-ids or text.

File: ui/pages/settings/contact-list-tab/edit-contact/edit-contact.component.js (hunk 22 above)
Comment (hmalik88) [reply ui/pages/settings/contact-list-tab/edit-contact/edit-contact.component.js:99]: I'll replace once we get the `BadgeIcon` in, thanks!

**Review by georgewrmarshall — APPROVED** [review]

**Review by FrederikBolding — APPROVED** [review]

File: ui/pages/confirmations/send/send-content/add-recipient/domain-input.component.js [hunk 24]
```diff
@@ -179,7 +179,7 @@ export default class DomainInput extends Component {
               <ButtonIcon
                 className="ens-input__wrapper__action-icon-button"
                 onClick={() => {
-                  if (userInput) {
+                  if (userInput.length > 0) {
```
Comment (jpuri) [inline ui/pages/confirmations/send/send-content/add-recipient/domain-input.component.js:182]: May be `if (userInput?.length > 0) {`

**Review by jpuri — APPROVED** [review]
